src/react/reconcilation.js

```diff
@@ -198,6 +199,9 @@ export class Reconciler {
         )
       )
     );
+    if (!checkIfRenderWasPure(effects, evaluatedRootNode)) {
+      return null;
```

Comment:
If the render was impure, we skip applying it's effects. So the effects are never used and should be restored.